django/db/models/query.py

```diff
@@ -635,6 +635,9 @@ def update(self, **kwargs):
         self._for_write = True
         query = self.query.clone(sql.UpdateQuery)
         query.add_update_values(kwargs)
+        # Clear any annotations so that they won't be present in subqueries
+        query.set_annotation_mask(None)
```

Comment:
It's unclear if `set_annotation_mask(None)` is necessary here. The tests still pass without it, and I included it to err on the side of caution in ensuring annotations were cleared, but I don't know the exact use case.

Can anyone think of a test which would cover `set_annotation_mask`?